src/cuda/backend.f90

```diff
@@ -415,21 +421,43 @@ subroutine tds_solve_dist(self, du, u, dirps, tdsops, blocks, threads)
 
    end subroutine tds_solve_dist
 
-   subroutine trans_x2y_cuda(self, u_y, v_y, w_y, u, v, w)
+   subroutine trans_x2y_cuda(self, u_y, u)
```

Comment:
Is code duplication needed here for performance reasons? Because I think it would be clearer, easier to maintain and less error prone if it was a single subroutine with something like:
```Fortran
trans_a2b_cuda(dir_a, dir_b, u_b, u)
integer, intent(in) :: dir_a
integer, intent(in) :: dir_b
...
select case (dir_a)
   case (dir_x)
     blocks = ...

end select
 ...
```

With `dir_x`, `dir_y` and `dir_z` parameters defined somewhere. That's just one approach, but we can have something else, like providing the content of blocks or threads as argument or in a separate function. I just feel like there is very few changes between the different functions and that doesn't really require duplication.

Reply:
At this level we can figure out a better way to handle transpose operation and it shouldn't have any effect on performance.
What I have in mind is having something like `transpose(u_, u, TR_XTOY)`, where we define all possible transposition directions as integer parameters somewhere. Then the backends can check this parameter and call the right function.